### PR TITLE
SDK upgrade

### PR DESCRIPTION
**Pull request by Jeday**
## Done

- [x] add wq hooks to react package
- [x] optimise useEthBalance
- [x] optimise useEthCost
- [x] add swr config passtrough 
- [x] add invariant to forbid using context hooks outside providers 
- [x] fix chainId invariant in web3react provider 
- [x] check in staking widger/reef-knot 
- [x] update tests 

## Note on SWR upgrade
I tried bumping SWR to v2 but it went terribly. Mostly crushed types and tests that rely on older swr behavior. Even minor bump up of swr will lead to type errors on build so be aware.

### CODE REVIEW

File: packages/react/src/hooks/useEthereumBalance.ts [hunk 1]
```diff
@@ -16,25 +17,22 @@ export const useEthereumBalance = (
     shouldFetch: !!mergedAccount,
     method: 'getBalance',
     params: [mergedAccount, 'latest'],
+    config,
   });
 
-  const updateBalance = useDebounceCallback(result.update);
-
-  const subscribeToUpdates = useCallback(() => {
+  useEffect(() => {
```
Comment (DiRaiks) [inline packages/react/src/hooks/useEthereumBalance.ts:23]: Let's revert debounce for the request

File: packages/react/src/hooks/useSDK.ts [hunk 2]
```diff
@@ -1,6 +1,9 @@
 import { useContext } from 'react';
 import { SDKContext, SDKContextValue } from '../context';
+import invariant from 'tiny-invariant';
 
 export const useSDK = (): SDKContextValue => {
-  return useContext(SDKContext);
+  const r = useContext(SDKContext);
```
Comment (DiRaiks) [inline packages/react/src/hooks/useSDK.ts:6]: Rename to valid const name

File: packages/react/src/hooks/useTokenBalance.ts [hunk 3]
```diff
@@ -27,13 +28,12 @@ export const useTokenBalance = (
     contract: contractRpc,
     method: 'balanceOf',
     params: [mergedAccount],
+    config,
   });
 
-  const updateBalance = useDebounceCallback(result.update);
-
-  const subscribeToUpdates = useCallback(() => {
+  useEffect(() => {
```
Comment (DiRaiks) [inline packages/react/src/hooks/useTokenBalance.ts:34]: Let`s revert debounce for the request

File: packages/react/src/hooks/useTotalSupply.ts [hunk 4]
```diff
@@ -21,13 +24,12 @@ export const useTotalSupply = (token: string): SWRResponse<BigNumber> => {
   const result = useContractSWR({
     contract: contractRpc,
     method: 'totalSupply',
+    config,
   });
 
-  const updateTotal = useDebounceCallback(result.update);
-
-  const subscribeToUpdates = useCallback(() => {
+  useEffect(() => {
```
Comment (DiRaiks) [inline packages/react/src/hooks/useTotalSupply.ts:30]: Let`s revert debounce for the request

File: packages/web3-react/src/hooks/useConnectors.ts [hunk 5]
```diff
@@ -1,6 +1,9 @@
 import { useContext } from 'react';
 import { ConnectorsContext, ConnectorsContextValue } from '../context';
+import invariant from 'tiny-invariant';
 
 export const useConnectors = (): ConnectorsContextValue => {
-  return useContext(ConnectorsContext);
+  const r = useContext(ConnectorsContext);
```
Comment (DiRaiks) [inline packages/web3-react/src/hooks/useConnectors.ts:6]: Use valid const name

**Review by DiRaiks — APPROVED** [review]
LGTM

**Review by Jabher — APPROVED** [review]

**Pull request merged**